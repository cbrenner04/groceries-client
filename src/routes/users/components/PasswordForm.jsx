import React from 'react';
import PropTypes from 'prop-types';

import { PasswordField } from '../../../components/FormFields';

const PasswordForm = props => (
  <form className="form" onSubmit={props.submissionHandler} autoComplete="off">
    <PasswordField
      name="password"
      label="Password"
      value={props.password}
      handleChange={props.passwordChangeHandler}
      placeholder="New password"
    />
    <PasswordField
      name="password-confirmation"
      label="Password confirmation"
      value={props.passwordConfirmation}
      handleChange={props.passwordConfirmationChangeHandler}
      placeholder="Confirm new password"
    />
    <button type="submit" className="btn btn-success btn-block">
      Set my password
    </button>
  </form>
);

PasswordForm.propTypes = {
  password: PropTypes.string.isRequired,
  passwordChangeHandler: PropTypes.func.isRequired,
  passwordConfirmation: PropTypes.string.isRequired,
  passwordConfirmationChangeHandler: PropTypes.func.isRequired,
  submissionHandler: PropTypes.func.isRequired,
};

export default PasswordForm;
