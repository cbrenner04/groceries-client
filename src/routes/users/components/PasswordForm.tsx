import React, { type ChangeEventHandler, type FormEventHandler } from 'react';
import { Form } from 'react-bootstrap';

import { PasswordField } from 'components/FormFields';
import FormSubmission from 'components/FormSubmission';

export interface IPasswordFormProps {
  password: string;
  passwordChangeHandler: ChangeEventHandler;
  passwordConfirmation: string;
  passwordConfirmationChangeHandler: ChangeEventHandler;
  submissionHandler: FormEventHandler;
}

const PasswordForm: React.FC<IPasswordFormProps> = (props): React.JSX.Element => (
  <Form onSubmit={props.submissionHandler} autoComplete="off" data-test-id="password-form">
    <PasswordField
      name="password"
      label="Password"
      value={props.password}
      handleChange={props.passwordChangeHandler}
      placeholder="New password"
    />
    <PasswordField
      name="password-confirmation"
      label="Password confirmation"
      value={props.passwordConfirmation}
      handleChange={props.passwordConfirmationChangeHandler}
      placeholder="Confirm new password"
    />
    <FormSubmission submitText="Set my password" />
  </Form>
);

export default PasswordForm;
