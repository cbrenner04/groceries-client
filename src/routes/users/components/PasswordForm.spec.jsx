import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import PasswordForm from './PasswordForm';

describe('PasswordForm', () => {
  let props;

  beforeEach(() => {
    props = {
      password: '',
      passwordChangeHandler: jest.fn(),
      passwordConfirmation: '',
      passwordConfirmationChangeHandler: jest.fn(),
      submissionHandler: jest.fn(),
    };
  });

  it('renders', () => {
    const { container } = render(<PasswordForm {...props} />);

    expect(container).toMatchSnapshot();
  });

  it('changes values', () => {
    const { getByLabelText } = render(<PasswordForm {...props} />);

    fireEvent.change(getByLabelText('Password'), { target: { value: 'foo' } });

    expect(props.passwordChangeHandler).toHaveBeenCalled();

    fireEvent.change(getByLabelText('Password confirmation'), { target: { value: 'foo' } });

    expect(props.passwordConfirmationChangeHandler).toHaveBeenCalled();
  });

  it('submits', () => {
    const { getByLabelText, getByTestId } = render(<PasswordForm {...props} />);

    fireEvent.change(getByLabelText('Password'), { target: { value: 'foo' } });
    fireEvent.change(getByLabelText('Password confirmation'), { target: { value: 'foo' } });
    fireEvent.submit(getByTestId('password-form'));

    expect(props.submissionHandler).toHaveBeenCalled();
  });
});
