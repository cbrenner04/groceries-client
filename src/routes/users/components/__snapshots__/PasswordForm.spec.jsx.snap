// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordForm renders 1`] = `
<div>
  <form
    autocomplete="off"
    class=""
    data-test-id="password-form"
  >
    <div
      class="mb-3"
    >
      <label
        class="form-label"
        for="password"
      >
        Password
      </label>
      <input
        autocomplete="off"
        class="form-control"
        id="password"
        name="password"
        placeholder="New password"
        type="password"
        value=""
      />
    </div>
    <div
      class="mb-3"
    >
      <label
        class="form-label"
        for="password-confirmation"
      >
        Password confirmation
      </label>
      <input
        autocomplete="off"
        class="form-control"
        id="password-confirmation"
        name="password-confirmation"
        placeholder="Confirm new password"
        type="password"
        value=""
      />
    </div>
    <div
      class="d-grid gap-2 mt-3"
    >
      <button
        class="btn btn-success"
        type="submit"
      >
        Set my password
      </button>
    </div>
  </form>
</div>
`;
